### DISCUSSION TITLE
Coponent making api call even when display="none"

### DISCUSSION THREAD

**Discussion by roker15** [chakra-ui/chakra-ui#6878] [Q&A] [answered]
I am using VStack to wrap component with conditional display

` display={selectedHeading === headings.id ? "block" : "none"}`

So component is not displayed but still it is making netowrk/ api call.

  

                <VStack                
                    display={selectedHeading === headings.id ? "block" : "none"}                                    
                  >
                   <ArticleCounter />                    
                   
                  </VStack>

There is api call inside   `<ArticleCounter />   ` which is being made even though `VStack  ` is hidden

**Comment by selbekk** [comment, marked as answer]
As long as the component renders, it will trigger any API  calls. Instead - stop it from rendering:

```tsx
<VStack>
  {selectedHeading === headings.id && (
    <ArticleCounter />
  )}
</VStack>
  ```
